javascript_nodejs/5.multiturn-prompts-bot/dialogs/mainDialog/index.js

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActionTypes, MessageFactory } = require('botbuilder');
+
+const { TextPrompt, NumberPrompt, DialogSet } = require('botbuilder-dialogs');
+
+const USER_NAME_PROP = 'user_name';
+const AGE_PROP = 'user_age';
+const WHO_ARE_YOU = 'who_are_you';
+const HELLO_USER = 'hello_user';
+
+class MainDialog {
+    /**
+     * 
+     * @param {Object} conversationState 
+     * @param {Object} userState 
+     */
+    constructor (conversationState, userState) {
+        var that = this;
+
+        // creates a new state accessor property. see https://aka.ms/about-bot-state-accessors to learn more about the bot state and state accessors 
+        this.conversationState = conversationState;
+        this.userState = userState;
+
+        this.userName = this.userState.createProperty(USER_NAME_PROP);
+        this.userAge = this.userState.createProperty(AGE_PROP);
+
+        this.dialogs = new DialogSet();
+        
+        // Add prompts
+        this.dialogs.add('textPrompt', new TextPrompt());
+        this.dialogs.add('numberPrompt', new NumberPrompt(async (context, value)=> {
+            if (value < 0) {
+                await context.sendActivity(`Your age can't be less than zero.`);
+                return undefined;
+            } else {
+                return value;
+            }
+        }));
+            
+        // Create a dialog that asks the user for their name.
+        this.dialogs.add(WHO_ARE_YOU, [
+            async function(dc) {
+                return await dc.prompt('textPrompt', `What is your name, human?`);
+            },
+            async function(dc, name_value) {
```

Comment:
This goes back to my previous comment about this using an old package, the second parameter is actually a `WaterfallStepContext`, not just the response from the user.